test/functional/Java12andUp/src/org/openj9/test/java_lang_invoke/Test_VarHandle.java

```diff
@@ -0,0 +1,162 @@
+            arrayTest2 = Helper.getArrayTest2();
+        } catch (Throwable e) {
+            Assert.fail("Test variables could not be initialized for Test_VarHandle.");
+        }
+    };
+
+    /*
+     * Test Java 12 API VarHandle.describeConstable() for instance field type
+     */
+    @Test(groups = { "level.sanity" })
+    public void testVarHandleDescribeConstableInstanceField() {
+        describeConstableTestGeneral("testVarHandleDescribeConstableInstanceField", instanceTest);
+    }
+
+
+    /*
+     * Test Java 12 API VarHandle.describeConstable() for static field type
+     */
+    @Test(groups = { "level.sanity" })
+    public void testVarHandleDescribeConstableStaticField() {
+        describeConstableTestGeneral("testVarHandleDescribeConstableStaticField", staticTest);
+    }
+
+    /*
+     * Test Java 12 API VarHandle.describeConstable() for array type
+     */
+    @Test(groups = { "level.sanity" })
+    public void testVarHandleDescribeConstableArray() {
+        describeConstableTestGeneral("testVarHandleDescribeConstableArray (1)", arrayTest);
+        describeConstableTestGeneral("testVarHandleDescribeConstableArray (2)", arrayTest2);
+    }
+
+    private void describeConstableTestGeneral(String testName, VarHandle handle) {
+        Optional<VarHandleDesc> optionalDesc = handle.describeConstable();
+        if (optionalDesc.isPresent()) {
+            VarHandleDesc desc = optionalDesc.get();
+
+            /* verify that descriptor can be resolved */
+            VarHandle resolvedHandle;
+            try {
+                resolvedHandle = (VarHandle)desc.resolveConstantDesc(MethodHandles.lookup());
+            } catch(ReflectiveOperationException e) {
+                Assert.fail(testName + ": ReflectiveOperationException thrown at VarHandleDesc.resolveConstantDesc() call");
+                return;
+            }
+            
+            String originalVarString = handle.varType().descriptorString();
+            String newVarString = resolvedHandle.varType().descriptorString();
+            logger.debug(testName + ": Descriptor of original class varType is: " + originalVarString + " descriptor of VarHandleDesc varType is: " + newVarString);
+            Assert.assertTrue(newVarString.equals(originalVarString));
+        } else {
+            Assert.fail("testVarHandleDescribeConstableInstanceField: No value present in Optional<VarHandleDesc>");
+        }
+    }
+
+    /*
+     * Test Java 12 API VarHandle.equals()
+     */
+    @Test(groups = { "level.sanity" })
+    public void testVarHandleEquals() {
+        logger.debug("testVarHandleHashCode: the same VarHandle should be equal to itself: " + instanceTest.equals(instanceTest));
+        Assert.assertTrue(instanceTest.equals(instanceTest));
+        logger.debug("testVarHandleHashCode: different VarHandle instances should not be equal: " +  !instanceTest.equals(staticTest));
+        Assert.assertTrue(! instanceTest.equals(staticTest));
```

Comment:
Can this be `assertEquals`?  here and elsewhere in this code?

Reply:
testng throws `error: method equals in class Object cannot be applied to given types`